pkg/scheduler/framework/plugins/imagelocality/image_locality.go

```diff
@@ -63,7 +63,16 @@ func (pl *ImageLocality) Score(ctx context.Context, state *framework.CycleState,
 	}
 	totalNumNodes := len(nodeInfos)
 
-	score := calculatePriority(sumImageScores(nodeInfo, pod.Spec.Containers, totalNumNodes), len(pod.Spec.Containers))
+	// Filtering containers with ImagePullPolicy different from Always
+	// Ones with Always will score 0 point either way
+	var notAlwaysContainers []v1.Container;
+	for _, container := range pod.Spec.Containers {
+		if container.ImagePullPolicy != "Always" {
+			notAlwaysContainers = append(notAlwaysContainers, container)
+		}
+	}
```

Comment:
in the worst case, where none of the containers have `Always`, we'll end up looping through the whole list twice (because [`sumImageScores`](https://github.com/kubernetes/kubernetes/blob/46f3e4d/pkg/scheduler/framework/plugins/imagelocality/image_locality.go#L97-L105) also loops through all the containers)

So what about updating the [check in `sumImageScores`](https://github.com/kubernetes/kubernetes/blob/46f3e4d/pkg/scheduler/framework/plugins/imagelocality/image_locality.go#L100) to just:
```go
if state, ok := nodeInfo.ImageStates[normalizedImageName(container.Image)]; ok && container.ImagePullPolicy != "Always" {
...
}
```

Reply:
Oh wow, I'm sorry, that's basically exactly what you had before I asked you to change it 🤦